File: java/src/main/java/com/myriad/wdl/model/processors/WdlFunctionProcessorBase.java
package com.myriad.wdl.model.processors;

import com.myriad.wdl.model.expressions.WdlFunctionCallOperation;

/** Base function-call processor that dispatches by {@link WdlFunctionCallOperation.WdlFunction}. */
public abstract class WdlFunctionProcessorBase implements WdlFunctionProcessor {

  @Override
  public final void processFunctionCall(WdlFunctionCallOperation functionCall) {
    if (functionCall == null) {
      return;
    }
    dispatchByFunction(functionCall);
  }
}

File: python/src/wdl_model/model/processors/wdl_function_processor_base.py
"""Default function-call dispatch helper for the Python WDL object model."""

from __future__ import annotations

from wdl_model.model.expressions import WdlFunctionCallOperation

from .wdl_function_processor import WdlFunctionProcessor


class WdlFunctionProcessorBase(WdlFunctionProcessor):
    """Base implementation that forwards a function call into ``dispatchByFunction``."""

    def processFunctionCall(
        self, functionCall: WdlFunctionCallOperation | None
    ) -> None:
        if functionCall is None:
            return
        self.dispatchByFunction(functionCall)
